dask_jobqueue/tests/test_jobqueue_core.py

```diff
@@ -413,3 +413,62 @@ def test_security(EnvSpecificCluster, loop):
         security=security,
     ) as cluster:
         assert "tls://" in job_script
+
+
+@pytest.mark.xfail_env({"htcondor": "#535 no shared filesystem in htcondor ci"})
+@pytest.mark.xfail_env({"slurm": "#535 no shared filesystem in slurm ci"})
+def test_security_temporary(EnvSpecificCluster, loop):
+    dirname = os.path.dirname(__file__)
+    with open(os.path.join(dirname, "key.pem"), "r") as file:
+        key = file.read()
```

Comment:
I've just some doubts here: what file are you reading? When was it created?

Reply:
OK, just seen this came from previous modifications. This certs were already there.

Reply:
yes the are part of the test suite. I could generate them however this adds a test dependency to `cryptography`

Reply:
It's OK that way I think. Except we don't test the generation though. I'm okay to add cryptography as a test dependency!

Reply:
I changed this so it to use `Security.temporary() `. This also better shows the most common use case I guess.